rust/automerge/src/query/seek_mark.rs

```diff
@@ -1,56 +1,45 @@
 use crate::marks::Mark;
 use crate::op_tree::OpSetMetadata;
-use crate::query::{QueryResult, TreeQuery};
-use crate::types::{Key, ListEncoding, Op, OpId, OpType};
+use crate::query::{ListState, OpTreeNode, QueryResult, TreeQuery};
+use crate::types::{ListEncoding, Op, OpId, OpType};
 use std::cmp::Ordering;
 use std::collections::HashMap;
 use std::fmt::Debug;
 
 #[derive(Debug, Clone, PartialEq)]
 pub(crate) struct SeekMark<'a> {
     /// the mark we are looking for
+    idx: ListState,
     id: OpId,
     end: usize,
-    encoding: ListEncoding,
     found: bool,
     mark_name: smol_str::SmolStr,
     next_mark: Option<Mark<'a>>,
-    pos: usize,
-    seen: usize,
-    last_seen: Option<Key>,
     super_marks: HashMap<OpId, smol_str::SmolStr>,
     pub(crate) marks: Vec<Mark<'a>>,
 }
 
 impl<'a> SeekMark<'a> {
     pub(crate) fn new(id: OpId, end: usize, encoding: ListEncoding) -> Self {
         SeekMark {
+            idx: ListState::new(encoding, usize::MAX),
```

Comment:
Yeah . These are the few functions where adding an `if` statement literally costs us 1% on a benchmark.  I'll take a usize::MAX over an Option<usize> just to avoid having to check if before <.  This gives the same result with fewer ops.